r/R/main.R

```diff
@@ -86,52 +87,23 @@ calc_K <- function(k, temp_c = 25, sal = 35, p_bar = NULL, magnesium = 0.0528171
     dat[, k_value := k_value * tot_to_sws_surface * check_pc * sws_to_tot_deep]
   }
 
-  # Calculate correction factor
-  if (tolower(method) == "myami") {
-    pymyami <- reticulate::import("pymyami")
-    Fcorr <- pymyami$calculate_seawater_correction(Sal = dat$sal, TempC = dat$temp_c, Mg = dat$magnesium, Ca = dat$calcium)
-    if (k %in% names(Fcorr)) {
-      KF <- as.numeric(Fcorr[[k]])
-      dat[, k_value := k_value * KF]
-    }
-  }
-  if (tolower(method) == "myami_polynomial") {
-    pymyami <- reticulate::import("pymyami")
-    Fcorr <- pymyami$approximate_seawater_correction(Sal = dat$sal, TempC = dat$temp_c, Mg = dat$magnesium, Ca = dat$calcium)
-    if (k %in% names(Fcorr)) {
-      KF <- as.numeric(Fcorr[[k]])
-      dat[, k_value := k_value * KF]
-    }
-  }
-  if (tolower(method) == "r_polynomial") {
-    poly_coefs <- rjson::fromJSON(file = system.file("coefficients/polynomial_coefficients.json", package = "kgen"))
-    if (k %in% names(poly_coefs)) {
-      # Calculate correction factors
-      dat[, row_id := .I]
-      dat[, KF := poly_coefs[[k]] %*% kgen_poly(sal = sal, temp_k = temp_k, magnesium = magnesium, calcium = calcium), by = row_id]
-      dat[, k_value := k_value * KF]
-    }
-  }
+  seawater_correction <- calc_seawater_correction(k=k, sal = sal, temp_c = temp_c, calcium = calcium, magnesium = magnesium)
+  dat[, k_value := k_value*seawater_correction]
 
   return(dat$k_value)
 }
 
 #' @title Calculate equilibrium constants for seawater
 #'
-#' @description Wrapper to calculate multiple stoichiometric equilibrium constants at given temperature, salinity, pressure and the concentration of magnesium, calcium, sulphate, and fluorine.
+#' @description Wrapper to calculate specified stoichiometric equilibrium constants at given temperature, salinity, pressure and the concentration of magnesium, calcium, sulphate, and fluorine.
 #'
 #' @author Dennis Mayk
 #'
 #' @inheritParams calc_K
 #' @param ks character vectors of Ks to be calculated e.g., c("K0", "K1")
 #' @return Data.table of specified Ks at the given conditions
 #' @export
```

Comment:
why do you want to export calc_K, calc_KS and calc_all_Ks? Those seem redundant.

Reply:
They are a little redundant.
Matlab and Python don't have (or aren't using) this concept of 'exporting' so all the functions we write will be available to the end user. Given it's possible to write `kgen.calc_K()`, `kgen.calc_Ks()` and `kgen.calc_all_Ks()` in Python and Matlab, I want it to be possible to do the same thing in R.

Reply:
Makes sense. But I wonder since R has the nice feature to define exported functions if we shouldn't use it. After all, very few people will move back and forth between languages but stick with one so we should make the programming as natural as possible for the respective language. If you still want to do that we will need to improve the docs to make it clear that these functions essentially do the same.

Reply:
Collaborators on a project are likely to have to fit in with whatever language the PI is familiar with, which might involve moving between languages.

This should be merged with the other thread as they're drifting into the same territory here, but the way I've written the Matlab isn't the 'natural' way to write Matlab, because the intention is to be easily comparable to the other languages, not to make a Matlab K generator then verify the output. Otherwise why did we write these K generators from scratch rather than just comparing the existing ones?